pkg/gadgets/audit-seccomp/tracer/bpf/audit-seccomp.c

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright (c) 2021 The Inspektor Gadget authors */
+
+/* This BPF program uses the GPL-restricted function bpf_probe_read*().
+ */
+
+#include <vmlinux/vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+
+#include "audit-seccomp.h"
+
+#include <gadgettracermanager/bpf-maps.h>
+
+/* The stack is limited, so use a map to build the event */
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, __u32);
+	__type(value, struct event);
+} tmp_event SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+} events SEC(".maps");
+
+SEC("kprobe/audit_seccomp")
+int kprobe__audit_seccomp(struct pt_regs *ctx)
+{
+	unsigned long syscall = PT_REGS_PARM1(ctx);
+	int code = PT_REGS_PARM3(ctx);
+
+	struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+	__u64 mntns_id = BPF_CORE_READ(task, nsproxy, mnt_ns, ns.inum);
+	if (mntns_id == 0) {
+		return 0;
+	}
+
+#ifdef WITH_FILTER
+	__u32 *found = bpf_map_lookup_elem(&filter, &mntns_id);
+	if (!found)
+		return 0;
+#endif
+
+	__u32 zero = 0;
+	struct event *event = bpf_map_lookup_elem(&tmp_event, &zero);
```

Comment:
Is there always a value corresponding to key 0?

Reply:
Yes, for BPF maps of type array, the size is constant and it is defined by `max_entries`. In this case, this is a array with `max_entries = 1`, so there is only one entry at index zero.

Since it is a per-cpu array, each cpu will have access to a different buffer at index zero, so if this BPF program runs in parallel on different cpus, each will use separate buffers.

Reply:
Oh sorry, I thought it was a map hence my question.
But my question is irrelevant for an array.